### ISSUE TITLE
Four fast-suite tests fail on macOS only: float32 ulp equality in msl_sparse_dft, one-line mesh count in probe_fed referee contract

### ISSUE THREAD

**Issue by bk-squared** [bk-squared/rfx#876]
Four tests in the default fast suite fail on macOS (Darwin 25.5, Apple Silicon, CPU JAX) at `main` 8b9f514 while Linux CI is green. Found while verifying the tier-4b test reorg (#856): the same four fail identically on `main` at the old paths, so they are pre-existing and platform-dependent, not reorg losses.

**1. `tests/test_msl_sparse_dft.py::test_cropped_extractor_reproduces_full_plane_s_end_to_end[uniform-x | uniform-y | nonuniform-x]`**
`np.testing.assert_array_equal(s_crop, s_full)` demands bitwise equality between the cropped and full-plane extractors. On macOS they differ by one float32 ulp on 4 of 12 elements:

```
Max absolute difference among violations: 1.4901161e-07
Max relative difference among violations: 9.638923e-07
[0, 0, 1]: 0.045018117874860764+0.04389047250151634j (ACTUAL) vs 0.04501805827021599+0.043890468776226044j (DESIRED)
```

This is summation-order / BLAS noise (Accelerate vs MKL/OpenBLAS), not an extractor defect. The claim the test makes ("cropped reproduces full plane") is true to 1e-6 relative. Proposal: keep the intent, replace the bitwise assert with `assert_allclose(rtol=4*np.finfo(np.float32).eps, atol=0)` and state in the docstring that the two paths are not required to be bitwise identical across BLAS backends. This is a tolerance *on a numerical-identity check between two code paths*, not on a physics gate, so it does not touch the accuracy contract.

**2. `tests/test_probe_fed_msl_referee_contract.py::test_plan_computes_the_post_smoothing_mesh_the_builder_hands_openems`**
```
assert 402 == 401
```
The plan's predicted post-smoothing mesh has one line more than the builder's on macOS. This one is worth understanding before touching: a one-line difference in a smoothed mesh is a floating-point tie in a line-placement comparison (a `<` vs `<=` on a value that lands on the boundary differently across platforms). Needs someone to dump both line sets and find the extra line; it may indicate the plan/builder pair is sensitive to a boundary tie that could also bite on Linux with a different compiler flag.

Not blocking anything. Weekly GPU suite (#852) runs on Linux and does not see these.

**Comment by bk-squared** [comment]
Five more host-dependent failures found by the tier-3b full-suite run on the same Mac (Darwin 25.5, CPU JAX). All five reproduce identically on the untouched tier-4b tip `3bafe2f`, so they are independent of the reorg. Node ids are the tier-4b paths.

**Linux-only tests without a platform skip** (`tests/unit/misc/test_weekly_rss_reporter.py`): `test_read_proc_vmhwm_mb_returns_a_positive_number_on_linux`, `test_read_proc_vmhwm_mb_agrees_with_independent_ru_maxrss_band`, `test_live_sampler_line_survives_sigkill_with_dash_s` — `assert None is not None` (no `/proc` on macOS), `'[rss-live] VmHWM' not in stdout`. Fix is a `pytest.mark.skipif(sys.platform != "linux")`; these are deployment tests for the weekly reporter, which only runs on Linux.

**Two that deserve a look before any skip is added**, since each asserts a bitwise invariant that holds on Linux and not here:
- `tests/unit/boundaries/test_boundary_spec_cpml_budget.py::test_result_saturates_once_the_budget_covers_the_narrowest_axis` — `budget=9 changed the fields although the grid and the active absorber are identical to budget=8` (field sha `275093fbad04…` vs `66667fb9d783…`). If the budget really is saturated, the two runs should be the same computation; a differing hash on one platform means either a non-deterministic reduction order in the CPML update on this backend, or the budget is not actually saturated and the test only passes on Linux by coincidence. The second reading would be a real gate defect.
- `tests/unit/grid/test_fourth_order_stencil.py::test_update_order2_state_byte_identical` — `jnp.array_equal` false for the order-2 stencil state on this host. Same question: platform reduction-order noise vs. an actual difference in the update path.

Suggested handling: skip the three `/proc` tests on non-Linux; for the two bitwise tests, dump the max |Δ| on macOS and decide whether the invariant is "same computation" (then the test should compare to a tolerance of a few ulp) or "same bytes" (then the CPML/stencil update has a platform-dependent branch to find).

**Comment by bk-squared** [comment]
## Root cause, per test (diagnosis only; no assertion touched)

**Environment.** This Mac: jax/jaxlib 0.10.2, numpy 2.4.6 (Accelerate), Python 3.11, arm64, plain CPU backend — **no jax-metal**. CI (`pr-tests.yml`): ubuntu, **Python 3.10**, unpinned `pip install -e ".[dev]"` → the last green run installed **jax 0.6.2 / numpy 2.2.6**. jax 0.6.2 is the last release supporting 3.10, so CI is silently frozen on a 2025 XLA.

**Decisive experiment.** A venv with CI's exact versions (jax 0.6.2, numpy 2.2.6) on this same Mac: CPML budget **passes**, stencil **passes**, sparse-DFT still fails (3/3), mesh still 402. Two failures are JAX/XLA-version effects, two are platform effects. All four numerical failures are deterministic run-to-run. Flag matrix: `--xla_cpu_enable_fast_math=false` 0/9; `JAX_ENABLE_X64=1` 3/9 (the sparse-DFT cases); `--xla_cpu_use_fusion_emitters=false` fixes CPML and sparse uniform-x/y only; matmul precision irrelevant (no matmul on any path).

1. **CPML budget saturation** — XLA fusion-emitter codegen (shape-dependent FMA contraction), **not an rfx defect**. Inputs byte-identical; only the z ψ-buffer depth differs (8 vs 9, an exact no-op layer with b=1, c=0, κ=1). Fields identical through step 5; first divergence at step 6, `ey`, 3 cells, 1 ulp. Emulation from the step-5 state: budget-9 matches `ey + cb*(d1*inv − d2*inv)`, budget-8 matches `ey + cb*fma(d1, inv, −d2*inv)`. The budget **is** saturated physically; the test passes on Linux because both programs happened to contract alike. Fix that keeps the standard: clamp `_axis_buffer_depths` to each axis's *active* absorber depth so budgets 8 and 9 compile to the same program, then assert program identity (lowered HLO / CPMLState shapes) plus the digest.

2. **Order-2 stencil byte identity** — XLA version: on 0.10.2 the eager reference `arr / dx` is rewritten to `arr * (1/dx)` (HLO shows `multiply`); the kernel is the IEEE-faithful side (0/336 mismatches vs numpy float32 `ex + cb*(d1/dx − d2/dx)`). ex 49/336, ey 60/336 differ, mostly 1–3 ulp, max 24 ulp; hx/hy identical. **Not an rfx defect.** Fix: compute the reference in numpy float32 and keep `array_equal` — a real bit-identity gate of the kernel against IEEE arithmetic.

3. **Cropped vs full-plane sparse-DFT** — arm64 codegen (shape-dependent FMA contraction in the fused complex accumulate); passes on x86 by coincidence. Ez accumulators and V bitwise identical in all 20 calls; cropped Hy/Hz accumulators (3,10,2) vs full (3,54,31) differ on 7–19/60 elements by 1 ulp → I differs in 1/4 calls by 8.5e-22 → S 4/12 by 1.49e-7. There is no reduction whose order differs (`update_dft_plane_probe` is elementwise; V and I are Python-loop sums in identical order), so no `jnp.sum` precision setting can fix it. Fix: keep the exact index-level witness (`test_sparse_monitor_matches_full_plane`, bitwise green here) and make the end-to-end S check a derived few-ulp `assert_allclose`, documented as a compiled-program identity check, not a physics tolerance.

4. **Post-smoothing mesh 402 vs 401** — macOS numpy build (FMA-contracted `np.arange`), amplified by the CSXCAD smoother's tie structure. Linux arm64 (Docker, glibc) → 401; emulated x86 → 401; macOS → 402 on numpy 2.2.6 and 2.4.6, so not ISA, not numpy version, not libm (±1-ulp log/pow perturbation leaves 402). `np.arange(y0, y1+dx/2, dx)` on macOS equals the single-rounding (fma) result at all 78 elements; 19/85 pre-smoothing lines differ by 1 ulp (4.4e-16 mm). The smoother's `argmin(dl)` over ulp-level widths diverges from call 4 (117 vs 114 calls) and its `np.ceil(rng/max_res)` ties (q = 1.0000000000000142 → 2 cells; 3.99999999999995 → 4) land on different intervals: a 1-ulp shift of a 2.6 mm line moves q by ~3.6e-14, larger than the tie margins. Not a `<`/`<=` issue; ceil under catastrophic cancellation. The same tie exists on Linux and any contracting build would flip it there. Fix: make the pre-smoothing lines bit-reproducible (`y0 + np.arange(n)*dx` as two explicit ufunc ops, or integer lattice × dx) and pin that line set as the committed cross-platform invariant. Do not widen 401.

5. **Weekly RSS reporter ×3** — Linux-only (`/proc`), `skipif(sys.platform != "linux")` is correct; the SIGKILL child did die by SIGKILL, only the stdout assertion failed.

**CI note.** Items 1 and 2 will fail on Linux too once CI runs Python ≥ 3.11 (jax ≥ 0.7). Proposal: add a Python 3.11 / current-jax job so version effects surface on the reference platform instead of only on developers' Macs.
